From 453b7dfc42abada57fd379965ea1866bab1eaf1e Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 30 Apr 2013 15:23:38 +0300
Subject: [PORT FROM MAIN] Telephony: Fix wakeup issues seen due to DataStall recovery

BZ: 28494

Telephony framework uses AlarmManager for recovery mechanism of data
stalled issue. Framework configures the ELAPSED_REALTIME_WAKEUP alarm
for every 6 minutes based on the data state.

This patch modifies the alarm to use ELAPSED_REALTIME which doesn't
wake the device up in sleep state

Change-Id: I419d12c5a8371a0501d77abec449ea69233a2da4
Orig-Change-Id: I7b47ff6e05dfc06f7e49bc859d6e2d746f6b06a3
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTrackerBase.java    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java b/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java
index a8050a6..8e481d7 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java
@@ -1622,7 +1622,7 @@ public abstract class DcTrackerBase extends Handler {
             intent.putExtra(DATA_STALL_ALARM_TAG_EXTRA, mDataStallAlarmTag);
             mDataStallAlarmIntent = PendingIntent.getBroadcast(mPhone.getContext(), 0, intent,
                     PendingIntent.FLAG_UPDATE_CURRENT);
-            mAlarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP,
+            mAlarmManager.set(AlarmManager.ELAPSED_REALTIME,
                     SystemClock.elapsedRealtime() + delayInMs, mDataStallAlarmIntent);
         } else {
             if (VDBG_STALL) {
-- 
1.7.4.1

